app/models/concerns/permalinkable.rb

```diff
@@ -9,7 +9,15 @@ module Permalinkable
 
   module ClassMethods
     def find_by_param!(param)
-      find_by_permalink!(param)
+      begin
+        find_by_permalink!(param)
+      rescue ActiveRecord::RecordNotFound
+        if param =~ /^\d+$/
```

Comment:
I only touched this method because it had a bug. The way it was implemented, we were never falling back to finding by id when the record was not found through the permalink.

But you're right. We shouldn't be using an exception in here. I'll try to review this in some other PR.